src/resources/fileutils.hpp

```diff
@@ -0,0 +1,74 @@
+#pragma once
+#include <dirent.h>
+#include <stdio.h>
+#include <cstring>
+#include <iterator>
+#include "filesystem_littlefs_bbf.h"
+
+#define log(severity, ...) _log_event(severity, log_component_find("Resources"), __VA_ARGS__)
```

Comment:
Do we need this `#define log()` anywhere in this header file? Or is it something common for the whole resources module? I'm kind of afraid about spoiling the rest of the code base with such a generic `#define log()`, especially when accidentally used by an inexperienced dev.

I'd rather hide the `#define` inside a cpp file.

Reply:
It is common for the resources module.

It is not spoiling the rest of the code base. This header file is considered private to the module and shall not be included by any other module.

I used here an approach I would maybe like us to use in the future. The public API of the `resources` module is under `include/resources`. There is no extra search path, so you should include it as `#include <resources/bootstrap.hpp>` (or with `""`, but that is for another discussion). The private header files are then inside the module's implementation, under `src/resources/` and included by the module just as a local include `#include "fileutils.hpp`

Reply:
Ok, if the header is not intended to be included anywhere besides the resources module, I'm fine with it.